src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs

```diff
@@ -311,10 +315,34 @@ public static bool TryGetLegacySettingElements(string runsettingsXml, out List<s
 
                     var childNodes = node.SelectChildren(XPathNodeType.Element);
 
+                    var legacySettingElements = new List<string>();
                     while (childNodes.MoveNext())
                     {
                         legacySettingElements.Add(childNodes.Current.Name);
                     }
+
+                    AddNodeToCI(@"/RunSettings/LegacySettings/Execution/TestTypeSpecific/UnitTestRunConfig/AssemblyResolution", runSettingsNavigator, legacySettingElements);
```

Comment:
Just a suggestion:
Create a string array for these and then do this foreach item in array.

Pros: 
a. No need to create a private method.
b. Better readability of nodes we are interested in.

Reply:
Done